pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/json/MutableJsonIndexImpl.java

```diff
@@ -291,6 +297,19 @@ private RoaringBitmap getMatchingFlattenedDocIds(Predicate predicate) {
     }
   }
 
+  private void record(ImmutableRoaringBitmap bitmap, FilterContext filter) {
+    InvocationRecording recording = Tracing.activeRecording();
+    if (recording.isEnabled()) {
+      recording.setColumnCardinality(_postingListMap.size());
+      recording.setFilter(FilterType.INDEX, describeJsonPredicate(filter.getPredicate()));
+      recording.setNumDocsMatchingAfterFilter(bitmap.getCardinality());
+    }
+  }
+
+  private static String describeJsonPredicate(Predicate predicate) {
+    return predicate == null ? "?" : predicate.getLhs().getIdentifier() + ":" + predicate.getType() + "?";
```

Comment:
Is this complete? This will return something like `myCol:JSON_MATCH?`

Reply:
It returns the column, the predicate type (usually EQ) and an argument. It must not capture PII from the value, and high cardinality data should be avoided. It is a low cardinality indicator of the predicate structure.

Reply:
I see. But this class is only for `JSON_MATCH` predicate, so should we move the recording into the `JsonMatchFilterOperator` (similar to `RANGE`, bitmap etc.)

Reply:
yes it can be moved there

Reply:
we lose the cardinality information but it's not very important